src/main/scala/spray/json/CollectionFormats.scala

```diff
@@ -23,9 +23,9 @@ trait CollectionFormats {
     * Supplies the JsonFormat for Lists.
    */
   implicit def listFormat[T :JsonFormat] = new RootJsonFormat[List[T]] {
-    def write(list: List[T]) = JsArray(list.map(_.toJson))
-    def read(value: JsValue) = value match {
-      case JsArray(elements) => elements.map(_.convertTo[T])
+    def write(list: List[T]) = JsArray(list.map(_.toJson).toVector)
```

Comment:
All these changes in here need to be performance verified. It would be easier if they'd be broken out into an extra PR.